internal/controller/clusterorder_controller.go

```diff
@@ -343,10 +345,18 @@ func (r *ClusterOrderReconciler) handleNoHostedCluster(ctx context.Context,
 
 	// only trigger webhook if the hostedcluster does not exist
 	if url := r.CreateClusterWebhook; url != "" {
-		if err := triggerWebHook(ctx, url, instance); err != nil {
+		remainingTime, err := triggerWebHook(ctx, url, instance, r.MinimumRequestInterval)
+		if err != nil {
 			log.Error(err, fmt.Sprintf("Failed to trigger webhook %s: %v", url, err))
 			return ctrl.Result{Requeue: true}, nil
 		}
+
+		// Verify if we are within the minimum request window
+		if remainingTime != 0 {
+			return ctrl.Result{RequeueAfter: remainingTime}, nil
+		}
+		log.Error(err, fmt.Sprintf("Failed to trigger webhook %s: %v", url, err))
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
how that requeue works internally? Asking because I would like to understand the impact of the requeue mechanism on the memory if we have a large among of items in it.

Reply:
It's a priority queue used by controller-runtime for a long time, and the same one where events from the k8s API server get queued in potentially high volume. This code path also includes automatic back-off. So I wouldn't worry too much about memory use here in general unless there's a specific concern.